### ISSUE TITLE
06. Semantics [ref. JP32 034]

### ISSUE THREAD

**Issue by AndrewSales** [Schematron/schematron-enhancement-proposals#56]
This clause lacks the most important part: a processing model for (1) choosing phases, patterns, and rules, (2) establishing rule contexts by evaluating context expressions, (3) evaluating assert and report elements in selected rules, and (4) compiling the SVRL report. At present, this clause defines ancillary mechanisms only.

**Comment by rjelliffe** [comment]
The initial version of Schematron specified the action of validation,
phases etc using predicate logic.

For the Rev 2 draft, the Japanese national body objected that it was not
correct, but could not point to how it was correct, or provide corrected
text. I could find no-one credible who understood the logic notation and
who was interested in contributing (I teeed up a local Computer Science
professor at the #1 Uni here, but they lost interest).  At which stage,
depression set in that I was in a Catch 22.

IIRC, I followed the Japanese request to move that text to a non-normative
note. But I still believe it was a mistake: the notional validation
operation of Schematron is a few nested case statements, and the standard
would be better if it gave a semi-formal statement (using pseudo-code or
logic).
